pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -150,12 +165,42 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		reasonsForWorkerPoolKubernetesUpdate[w.Name] = reasonForWorkerPoolKubernetesUpdate
 	}
 
-	maintainOperation(shoot)
-	maintainTasks(shoot, r.Config)
+	for workerPool, reason := range reasonsForWorkerPoolKubernetesUpdate {
+		operation := fmt.Sprintf("For worker pool %s : %s", workerPool, reason)
+		operations = append(operations, operation)
+	}
+
+	operation := maintainOperation(shoot)
+	if operation != "" {
+		operations = append([]string{operation}, operations...)
+	}
+
+	tasks := maintainTasks(shoot, r.Config)
+	operations = append(tasks, operations...)
+
+	now = r.Clock.Now()
+	patch := client.MergeFrom(newShoot.DeepCopy())
```

Comment:
```suggestion
	patch := client.MergeFrom(shoot.DeepCopy())
```

Reply:
OK, understood. Not quite intuitive, though. It would be more clear if the various `maintain*` functions rather return mutation functions `func(*gardencorev1beta1.Shoot)` instead of directly modifying the object. Then we can call those at the end before the `Update()` call and don't need this `newShoot`. WDYT? I don't insist on it for now, just an idea.

Reply:
Using mutate function for worker pool Kubernetes version is a bit to much overdo, so I will skip this.